compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1454,6 +1495,15 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
                 break;
         }
 
+        switch (currentCtx) {
+            case TABLE_CONSTRUCTOR:
+            case QUERY_EXPRESSION:
+                switchContext(currentCtx);
```

Comment:
We can switch the context at the point where we give `QUERY_EXPRESSION` / `TABLE_CONSTRUCTOR` as the next rule.

Not a big issue though, just thought it would be slightly clear for a reader.

Reply:
```
private static final ParserRuleContext[] TABLE_CONSTRUCTOR_OR_QUERY_START =
            { ParserRuleContext.TABLE_KEYWORD, ParserRuleContext.STREAM_KEYWORD, ParserRuleContext.QUERY_EXPRESSION };

    private static final ParserRuleContext[] TABLE_CONSTRUCTOR_OR_QUERY_RHS =
            { ParserRuleContext.TABLE_CONSTRUCTOR, ParserRuleContext.QUERY_EXPRESSION };
```

Actually those two ctx are not called directly. They are called inside above arrays ?

Reply:
ack. Then its OK to keep it.